hydrocarbon_problem/api/aspen_api.py

```diff
@@ -0,0 +1,146 @@
+import numpy as np
+# from sqlalchemy import column
+from hydrocarbon_problem.api.Simulation import Simulation
+
+from hydrocarbon_problem.api.api_base import BaseAspenDistillationAPI
+from hydrocarbon_problem.api.types_ import StreamSpecification, ColumnInputSpecification, \
+    ColumnOutputSpecification, ProductSpecification, PerCompoundProperty
+
+PATH = 'C:/Users/s2399016/Documents/Aspen-RL_v2/Aspen-RL/hydrocarbon_problem/AspenSimulation/HydrocarbonMixture.bkp'
+
+
+class AspenAPI(BaseAspenDistillationAPI):
+    def __init__(self):
+        self._flowsheet: Simulation = Simulation(PATH=PATH, VISIBILITY=True)
+        self._feed_name: str = "S1"
+        self._tops_name: str = "S2"
+        self._bottoms_name: str = "S3"
+        self._name_to_aspen_name = PerCompoundProperty(ethane="ETHANE",
+                                                       propane="PROPANE",
+                                                       isobutane="I-BUTANE",
+                                                       n_butane="N-BUTANE",
+                                                       isopentane="I-PENTAN",
+                                                       n_pentane="N-PENTAN")
+        
+    def set_input_stream_specification(self, stream_specification: StreamSpecification) -> None:
+        """Sets the input stream to a column to fit the stream specification"""
+        # Defining the Thermodynamic Properties
+        self._flowsheet.STRM_Temperature(self._feed_name,  stream_specification.temperature)
+        self._flowsheet.STRM_Pressure(self._feed_name, stream_specification.pressure)
+        # Defining the Stream Composition for the Feed
+        self._flowsheet.STRM_Flowrate(self._feed_name, self._name_to_aspen_name.ethane,     stream_specification.molar_flows.ethane)
+        self._flowsheet.STRM_Flowrate(self._feed_name, self._name_to_aspen_name.propane,    stream_specification.molar_flows.propane)
+        self._flowsheet.STRM_Flowrate(self._feed_name, self._name_to_aspen_name.isobutane,  stream_specification.molar_flows.isobutane)
+        self._flowsheet.STRM_Flowrate(self._feed_name, self._name_to_aspen_name.n_butane,   stream_specification.molar_flows.n_butane)
+        self._flowsheet.STRM_Flowrate(self._feed_name, self._name_to_aspen_name.isopentane, stream_specification.molar_flows.isopentane)
+        self._flowsheet.STRM_Flowrate(self._feed_name, self._name_to_aspen_name.n_pentane,  stream_specification.molar_flows.n_pentane)
+
+    def get_output_stream_specifications(self) -> Tuple[StreamSpecification, StreamSpecification]:
+        # Getting the physical values of Top streams
+        tops_temperature = self._flowsheet.STRM_Get_Temperature(self._tops_name)
+        tops_pressure = self._flowsheet.STRM_Get_Pressure(self._tops_name)
+        # Acquiring the outputs out of the Distillate (Top Stream)
+        tops_ethane = self._flowsheet.STRM_Get_Outputs(self._tops_name, self._name_to_aspen_name.ethane)
```

Comment:
Same here - can we retrieve all the stream info in one call to aspen? No problem if not - just checking as this will be faster.